### ISSUE TITLE
How can I import FullPage component properly?

### ISSUE THREAD

**Issue by yangkean** [alvarotrigo/vue-fullpage.js#4]
Hello, alvarotrigo. I use vue-fullpage.js in my project and follow your `Basic usage` guide. But I always get errors.

Following is my code:

### index.html
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>demo</title>
    <link rel="stylesheet" href="./static/fullpage.js/jquery.fullpage.min.css">
  </head>
  <body>
    <div id="app"></div>
    <script src="./static/jquery/jquery.min.js"></script>
    <script src="./static/fullpage.js/jquery.fullpage.min.js"></script>
    <script src="/dist/build.js"></script>
  </body>
</html>
```

### App.vue
```vue
<template>
  <full-page :options="options">
      <div class="section">
        section 1
      </div>
      <div class="section">
        section 2
      </div>
    </full-page>
</template>

<script>
import FullPage from 'vue-fullpage.js/src/FullPage';
import fullPageMixin from 'vue-fullpage.js/src/fullPageMixin';

export default {
  name: 'app',

  mixins: [fullPageMixin],

  components: {
    FullPage,
  },

  data() {
    return {
      options: {},
    };
  },
}
</script>

<style scoped>

</style>
```
then it will cause:
```bash
Module build failed: Error: Couldn't find preset "es2015" relative to directory "/path/to/node_modules/vue-fullpage.js"
```

I just don't want to use `Vue.component`. Can you tell me what I am doing wrong? Thanks a lot.

**Comment by alvarotrigo** [comment]
I'm not sure how you have your TypeScript definitions configured, but have you tried adding the `.vue` extension in `'vue-fullpage.js/src/FullPage';`?

```javascript
import FullPage from 'vue-fullpage.js/src/FullPage.vue';
import fullPageMixin from 'vue-fullpage.js/src/fullPageMixin';
```

Also, have you installed all the node modules required for the build?
Did you do `npm install`? See [this](https://stackoverflow.com/a/40188652/1081396).

**Comment by alvarotrigo** [comment]
You can try to run the demo examples yourself. 
Do npm install and then run:

```shell
npm run-script demo1
```

If that's working fine, then try to figure out where's the difference with your implementation.

**Comment by Kxrr** [comment]
https://github.com/JeffreyWay/laravel-mix/issues/158#issuecomment-275108320 will solve the issue, but after this, I meet something new...

**Comment by alvarotrigo** [comment]
@Kxrr the file `.babelrc` is present [here on the project](https://github.com/alvarotrigo/vue-fullpage.js/blob/master/.babelrc).

**Comment by alvarotrigo** [comment]
@yangkean I assume you fixed the issue?

**Comment by yangkean** [comment]
@alvarotrigo , thanks for your reply. I am so sorry that I am too busy finishing a project recently to check my emails. I will try your advice later and you can close this issue first. If there is still something wrong with it, I will open it again and give you feedback. By the way, your lib is really awesome! Thanks again!

**Comment by alvarotrigo** [comment]
I'll wait for you :)

**Comment by yangkean** [comment]
@alvarotrigo , I have figured it out!!!

There are three points breaking my demo.

1. I must run `npm i` in `vue-fullpage.js` package in `node_modules`. It may be inconvenient if I have to install dependencies of `vue-fullpage.js` before I use `vue-fullpage.js/src/FullPage` and `vue-fullpage.js/src/fullPageMixin`.
2. I should not use `webpack.ProvidePlugin` to define `jQuery` as a global constants when I have include `jQuery` in `<script>`.
3. When I define rules for js files in `webpack.config.js`, I should exclude node_modules but `vue-fullpage.js/src`, just like `exclude: /node_modules\/(?!vue-fullpage\.js\/src)/`. See [#2031](https://github.com/webpack/webpack/issues/2031#issuecomment-244921229).

Following are all my files that can work well:

**./src/App.vue**
```vue
<template>
  <full-page :options="options">
    <div class="section">
      section 1
    </div>
    <div class="section">
      section 2
    </div>
    <div class="section">
      section 3
    </div>
  </full-page>
</template>

<script>
import FullPage from 'vue-fullpage.js/src/FullPage';
import fullPageMixin from 'vue-fullpage.js/src/fullPageMixin';

export default {
  name: 'app',

  mixins: [fullPageMixin],

  components: {
    FullPage,
  },

  data() {
    return {
      options: {
        scrollBar: false,
        navigation: true,
        anchors: ['page1', 'page2', 'page3'],
        sectionsColor: ['#41b883', '#ff5f45', '#0798ec', '#fec401', '#1bcee6', '#ee1a59', '#2c3e4f', '#ba5be9', '#b4b8ab'],
      },
    };
  },
}
</script>

<style scoped>

</style>
```

**./src/main.js**
```js
import Vue from 'vue'
import App from './App.vue'

new Vue({
  el: '#app',
  render: h => h(App)
})
```

**./.babelrc**
```json
{
  "presets": [
    ["env", { "modules": false }]
  ]
}
```

**./index.html**
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>wechat-ad-restore</title>
    <link href="https://cdnjs.cloudflare.com/ajax/libs/fullPage.js/2.9.5/jquery.fullpage.min.css" rel="stylesheet">
  </head>
  <body>
    <div id="app"></div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fullPage.js/2.9.5/jquery.fullpage.js"></script>
    <script src="/dist/build.js"></script>
  </body>
</html>
```

**./package.json**
```json
{
  "name": "vue-fullpage.js-demo",
  "description": "demo",
  "version": "1.0.0",
  "author": "[Sam Yang] <[samyangcoder@gmail.com]>",
  "scripts": {
    "dev": "cross-env NODE_ENV=development webpack-dev-server --open --hot"
  },
  "dependencies": {
    "vue": "^2.5.11",
    "vue-fullpage.js": "0.0.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not ie <= 8"
  ],
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.2",
    "babel-preset-env": "^1.6.0",
    "cross-env": "^5.0.5",
    "css-loader": "^0.28.9",
    "file-loader": "^1.1.4",
    "vue-loader": "^13.0.5",
    "vue-template-compiler": "^2.4.4",
    "webpack": "^3.6.0",
    "webpack-dev-server": "^2.9.1"
  }
}
```

**./webpack.config.js**
```js
const path = require('path');

module.exports = {
  entry: [
    './src/main.js',
  ],
  output: {
    path: path.resolve(__dirname, './dist'),
    publicPath: '/dist/',
    filename: 'build.js'
  },
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          'vue-style-loader',
          'css-loader'
        ],
      },
      {
        test: /\.vue$/,
        loader: 'vue-loader',
      },
      {
        test: /\.js$/,
        loader: 'babel-loader',
        exclude: /node_modules\/(?!vue-fullpage\.js\/src)/,
      },
    ]
  },
  resolve: {
    alias: {
      'vue$': 'vue/dist/vue.esm.js'
    },
    extensions: ['*', '.js', '.vue', '.json']
  },
  devtool: '#eval-source-map'
};
```

**Comment by alvarotrigo** [comment]
Glad you found a solution!
If you believe that something should be changed on the docs or the project let me know!

I'll close the issue by now.

**Comment by yangkean** [comment]
Yep! I think it is more convenient to add `node_modules` folder to `vue-fullpage.js` [npm package](https://www.npmjs.com/package/vue-fullpage.js). Thanks a ton! @alvarotrigo

**Comment by alvarotrigo** [comment]
@yangkean why not just telling them to use `npm install`? All modules are declared in package.json as dev-dependencies.

**Comment by yangkean** [comment]
@alvarotrigo Yes, what you said is right. But before import `FullPage` and `fullPageMixin` someone should install `vue-fullpage.js` first, then `cd` into `vue-fullpage.js` folder which is in `node_modules` folder and then run `npm i`, instead of importing and using them directly like simple Vue.js components. Just my personal opinion, feel free to consider it. Thanks anyway! :)

**Comment by alvarotrigo** [comment]
@yangkean I'm not sure to follow you.

You first said you had the following error:

> Module build failed: Error: Couldn't find preset "es2015" relative to directory "/path/to/node_modules/vue-fullpage.js"

But... doing **what** exactly? Which command?

More than including the node_modules folder (that it seems no other Vue component does), it might be an issue of adding dependencies to the [package file](https://github.com/alvarotrigo/vue-fullpage.js/blob/master/package.json) instead of `devDependencies`.

**Comment by alvarotrigo** [comment]
@yangkean I don't think it needs to add `babel-preset-es2015`as a dependency, you can take a look at other vue components like [this one](https://github.com/ratiw/vue-table), and it is also placed in the devDependencies folder.

**Comment by yangkean** [comment]
@alvarotrigo Oh, how stupid I am! I know why I have to install devDependencies of `vue-fullpage.js`. Because of [babelrc](https://github.com/alvarotrigo/vue-fullpage.js/blob/master/.babelrc) file in `vue-fullpage.js` package, I need to install `babel-preset-es2015` and `babel-plugin-transform-object-rest-spread` as devDependencies under my root project folder so that the vue file and js file under **vue-fullpage.js/src/** can be transpiled correctly. I don't install them under my root project folder before so the following error happened:

> Module build failed: Error: Couldn't find preset "es2015" relative to directory "/path/to/node_modules/vue-fullpage.js"

my new package.json (**./package.json**) file is:
```json
{
  "name": "vue-fullpage.js-demo",
  "description": "demo",
  "version": "1.0.0",
  "author": "[Sam Yang] <[samyangcoder@gmail.com]>",
  "scripts": {
    "dev": "cross-env NODE_ENV=development webpack-dev-server --open --hot"
  },
  "dependencies": {
    "vue": "^2.5.11",
    "vue-fullpage.js": "0.0.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not ie <= 8"
  ],
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.2",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-preset-env": "^1.6.0",
    "babel-preset-es2015": "^6.24.1",
    "cross-env": "^5.0.5",
    "css-loader": "^0.28.9",
    "file-loader": "^1.1.4",
    "vue-loader": "^13.0.5",
    "vue-template-compiler": "^2.4.4",
    "webpack": "^3.6.0",
    "webpack-dev-server": "^2.9.1"
  }
}
```
It doesn't need to run `npm i` in `vue-fullpage.js` package in `node_modules` in fact. Instead, just install `babel-preset-es2015` and `babel-plugin-transform-object-rest-spread` under root project folder and everything will work well!! So sorry to disturb you much.

**Comment by DigitalWheelie** [comment]
This is similar to a problem I'm having.

I no longer get the `es2015` error, but now I'm getting:

`./node_modules/vue-fullpage.js/src/fullPageMixin.js`
`Module parse failed: Unexpected token (19:4)`
`You may need an appropriate loader to handle this file type.`
`|   methods: {`
`|     // Methods of Full Page`
`|     ...constants.METHODS.reduce((methods, method) => {`
`|       return {`
`|         ...methods,`
` @ ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/App.vue 43:0-62`
` @ ./src/App.vue`
` @ ./src/main.js`
` @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/main.js`

Any suggestions?

**Comment by alvarotrigo** [comment]
See the [webpack configuration](https://github.com/alvarotrigo/vue-fullpage.js/blob/master/webpack.config.build.js) used to compile those files for the demos.
Also check the [package.json file](https://github.com/alvarotrigo/vue-fullpage.js/blob/master/package.json) and see the dev dependencies.

**Comment by akinhwan** [comment]
what I learned from this that helped solve my issues with module build error, was that I had to include the script tag links in the index.html for cdn.cloudfare links to jquery min and jquery.fullpage.js...whew

**Comment by cloudsoh** [comment]
I'm using vue webpack template and I face this error
`error  in ./node_modules/vue-fullpage.js/src/fullPageMixin.js

Module parse failed: Unexpected token (19:4)
You may need an appropriate loader to handle this file type.
|   methods: {
|     // Methods of Full Page
|     ...constants.METHODS.reduce((methods, method) => {
|       return {
|         ...methods,

 @ ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/App.vue 12:0-65
 @ ./src/App.vue
 @ ./src/main.js
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/main.js
`
I see you asked us to look at the webpack configuration and package.json, but I couldn't fix it. Please help me :pray:

**Comment by yangkean** [comment]
@cloudsoh As the error says, you need to install `babel-preset-es2015` and `babel-plugin-transform-object-rest-spread` as devDependencies under your root project folder to support ES6 features. Try [my solution](https://github.com/alvarotrigo/vue-fullpage.js/issues/4#issuecomment-366520470).

**Comment by cloudsoh** [comment]
@yangkean Is there anything more to do after install these 2? I couldn't fix it after installing them.
I think the vue-cli has included the ES6 out of the box. I can use ES6 anywhere in my project. However when I include the fullPageMixin the error comes out.

**Comment by cloudsoh** [comment]
@DigitalWheelie have you fixed your problem? :rofl:

**Comment by DigitalWheelie** [comment]
Kind of. Instead of trying to use full;page.js w/ Webpack I'm using it "flat." That works for me. I tried yangkean's solution, but still had the problem. My guess is it's something to do w/ the webpack-simple template, but it's just a guess.

**Comment by alvarotrigo** [comment]
This should be now fixed on the latest version 0.0.2
